File: announcements/api/models/announcements.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Coordinates struct {
	Lat  float64 `json:"lat" bson:"lat"`
	Long float64 `json:"long" bson:"long"`
}

type Announcement struct {
	Id          primitive.ObjectID `bson:"_id"`
	UserID      string             `json:"userID" bson:"userID"`
	Category    string             `json:"category" bson:"category"`
	Description string             `json:"description" bson:"description"`
	Location    Coordinates        `json:"location" bson:"location"`
}

File: yellowfy/lib/models/announcements.dart
class Coordinate {
  final double latitude;
  final double longitude;
  Coordinate({required this.latitude, required this.longitude});
}

class Announcement {
  final String id;
  final String userId;
  final String category;
  final String description;
  final Coordinate coordinates;
  final String name;
  Announcement({
    required this.id,
    required this.userId,
    required this.category,
    required this.description,
    required this.coordinates,
    required this.name,
  });
}
